src/libNode/Node.cpp

```diff
@@ -1668,3 +1669,42 @@ std::string Node::GetActionString(Action action) const {
              ? "Unknown"
              : ActionStrings.at(action);
 }
+
+/*static*/ bool Node::GetDSLeaderPeer(const BlockLink& lastBlockLink,
+                                      const DSBlock& latestDSBlock,
+                                      const DequeOfDSNode& dsCommittee,
+                                      const uint64_t epochNumber,
+                                      Peer& dsLeaderPeer) {
+  const auto& blocktype = get<BlockLinkIndex::BLOCKTYPE>(lastBlockLink);
+  if (blocktype == BlockType::DS) {
+    uint16_t lastBlockHash = 0;
+    // To cater for boostrap of blockchain. The zero and first epoch the DS
+    // leader is at index0
+    if (latestDSBlock.GetHeader().GetBlockNum() > 1) {
+      lastBlockHash = DataConversion::charArrTo16Bits(
+          latestDSBlock.GetHeader().GetHashForRandom().asBytes());
+    }
+
+    uint32_t leader_id = 0;
+    if (!GUARD_MODE) {
+      leader_id = lastBlockHash % dsCommittee.size();
+    } else {
+      leader_id = lastBlockHash % Guard::GetInstance().GetNumOfDSGuard();
+    }
+    dsLeaderPeer = dsCommittee.at(leader_id).second;
+    LOG_EPOCH(INFO, to_string(epochNumber).c_str(),
+              "lastBlockHash " << lastBlockHash << ", current ds leader id "
+                               << leader_id << ", Peer "
+                               << std::string(dsLeaderPeer));
```

Comment:
Do we need tostring() for printing peer?

Reply:
No need, i fixed it already.